### ISSUE TITLE
Handle module name - source file name mismatch and print helpful message to the user

### ISSUE THREAD

**Issue by milancurcic** [fortran-lang/fpm#129]
See #128. Currently fpm does not help the user figure out what went wrong if the module name / file name convention is not followed. fpm should detect this and print a helpful message to the user, for example:

```
fpm error: I noticed that you defined a module `my` in file `src/mx.f90`. The module name must match the source (path) name. Either rename your module to `mx`, or place it in `src/my.f90`.
```
or similar.

**Comment by awvwgk** [comment]
This might be fixed with #213.

**Comment by everythingfunctional** [comment]
Correct, the bootstrap version is no longer enforcing any match between module and source file name.

**Comment by awvwgk** [comment]
Closing this issue as resolved by #213.
